### PR TITLE
Add links regarding Spotless integration for gradle/maven users

### PR DESCRIPTION
**Pull request by nedtwigg**
~~If there's a better place to suggest this edit, happy to move it around.  I know Gradle is a niche tool in the Python world, happy to be somewhere quieter than the main README, but it seems that the README gets sliced up into smaller pages somehow?~~ aha, your CI helped me understand.  I think it's in the right place now :)

[Spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle) is a code formatting plugin.  It's main value adds are:

- it lets you compose multiple formatters (e.g. black + a license header that slurps copyright info from git history)
- it fixes lots of common bugs in formatters ([idempotence](https://github.com/diffplug/spotless/blob/main/PADDEDCELL.md), misconfigured [encodings](https://github.com/diffplug/spotless/blob/main/PADDEDCELL.md), [fighting with git over line endings](https://github.com/diffplug/spotless/tree/main/plugin-gradle#line-endings-and-encodings-invisible-stuff))
- your format gets really fast thanks to incremental build, local and remote build cache, etc.
- you can [ratchet your formatting](https://github.com/diffplug/spotless/tree/main/plugin-gradle#ratchet) so there's never a "format the world" commit
- it's polyglot ([prettier](https://github.com/diffplug/spotless/tree/main/plugin-gradle#prettier), [clang-format](https://github.com/diffplug/spotless/tree/main/plugin-gradle#clang-format), tons of JVM formatters too)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -406,6 +406,22 @@ jobs:
       - uses: psf/black@stable
 ```
 
+## Spotless
```
Comment (JelleZijlstra) [inline README.md:409]: A separate section for a single tool that's fairly obscure to the Python community seems like overkill. Maybe we could mention this in the "editor integrations" section?

File: README.md [hunk 2]
```diff
@@ -406,6 +406,22 @@ jobs:
...
+
+If you happen to be using Gradle, the
+[Spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle) plugin has
+support for black. Spotless gives you the option to
+
+- [ratchet](https://github.com/diffplug/spotless/tree/main/plugin-gradle#ratchet) your
+  formatting so that only files which have changed since `origin/main` get formatted
+- slurp copyright info from
+  [git history](https://github.com/diffplug/spotless/tree/main/plugin-gradle#license-header)
+- seamlessly integrate with
```
Comment (JelleZijlstra) [inline README.md:419]: Integrations with formatters for other languages don't feel relevant to Black's documentation.

**Comment by nedtwigg** [comment]
Got the precommit hooks installed, which fixed my format violations (ironic!).  The experience I just had, as a new prospective contributor, was:

- CI error message indicates pre-commit is the cause
- read `CONTRIBUTING.md` more carefully
- install and fiddle with pre-commit

I love the QC, and it's very cool how pre-commit orchestrates tools from multiple ecosystems (e.g. prettier).  This workflow is exactly what Spotless does for Gradle projects, it just does it inline with the user experience that Gradle devs are already used to, where the user experience would have been this:

```console
bot@ci repo% ./gradlew build
:spotlessJavaCheck FAILED
  The following files had format violations:
  src\main\java\com\diffplug\gradle\spotless\FormatExtension.java
    -\t\t····if·(targets.length·==·0)·{
    +\t\tif·(targets.length·==·0)·{
  Run './gradlew spotlessApply' to fix these violations.
user@machine repo % ./gradlew spotlessApply
:spotlessApply
BUILD SUCCESSFUL
bot@ci repo % ./gradlew build
BUILD SUCCESSFUL
```

Convergent evolution from a different ecosystem :)

**Comment by nedtwigg** [comment]
All tests passed except `Primer / build (3.8, macOS-latest)` which looks to be flaky.  I'll go dormant to stop creating noise, happy to push an empty commit to trigger a rerun or respond to any other requests.

File: README.md (hunk 1 above)
Comment (nedtwigg) [reply README.md:409]: Sure, I'll move it there :). I was looking for an "integrations" or "build tools" section.  We don't quite fit "editor integrations", but we're pretty close.

File: README.md (hunk 2 above)
Comment (nedtwigg) [reply README.md:419]: As I see it, someone who comes to your website is saying "I want to stop caring about formatting", not "I want to stop caring about formatting just my python, but keep caring about my html templates, c code,  and markdown documentation".  So I think it's relevant that if you use Spotless as your integration layer, you only have one interface to worry about, versus "I'll use project.toml to format my python, precommit for my docs, and a shell-script to clang-format for my c-code."

The fact that this project relies on prettier for formatting its docs is a good example that the people who bother to use one autoformatter usually want others, and it's nice to be able to configure them all from one place, which is the whole point of Spotless.

One argument for keeping Spotless as a peer to VCS is that it offers unique features related to slurping info from git history, which can be valuable for projects with strict legal requirements, especially polyglot projects.  Regardless, I'll move Spotless to editor integration, and you can pick whichever commit you prefer :)

**Comment by ambv** [comment]
> Got the precommit hooks installed, which fixed my format violations (ironic!)

Yeah, admittedly this flow is not super great when editing on the Web, as I just found out myself 😅

**Comment by ambv** [comment]
Thanks! ✨ 🍰 ✨

**Comment by nedtwigg** [comment]
Thank you!

**Pull request merged**